--- a/net/minecraft/world/level/redstone/NeighborUpdater.java
+++ b/net/minecraft/world/level/redstone/NeighborUpdater.java
@@ -50,19 +_,36 @@
         // Paper end - Add source block to BlockPhysicsEvent
         try {
             // CraftBukkit start
-            org.bukkit.event.block.BlockPhysicsEvent event = new org.bukkit.event.block.BlockPhysicsEvent(org.bukkit.craftbukkit.block.CraftBlock.at(level, pos), org.bukkit.craftbukkit.block.data.CraftBlockData.fromData(state), org.bukkit.craftbukkit.block.CraftBlock.at(level, sourcePos)); // Paper - Add source block to BlockPhysicsEvent
-            level.getCraftServer().getPluginManager().callEvent(event);
+            if (org.bukkit.event.block.BlockPhysicsEvent.getHandlerList().getRegisteredListeners().length > 0) { // Canvas - Skip BlockPhysicsEvent if no listeners
+                org.bukkit.event.block.BlockPhysicsEvent event = new org.bukkit.event.block.BlockPhysicsEvent(org.bukkit.craftbukkit.block.CraftBlock.at(level, pos), org.bukkit.craftbukkit.block.data.CraftBlockData.fromData(state), org.bukkit.craftbukkit.block.CraftBlock.at(level, sourcePos)); // Paper - Add source block to BlockPhysicsEvent
+                level.getCraftServer().getPluginManager().callEvent(event);
 
-            if (event.isCancelled()) {
-                return;
-            }
+                if (event.isCancelled()) {
+                    return;
+                }
+            } // Canvas - Skip BlockPhysicsEvent if no listeners
             // CraftBukkit end
             state.handleNeighborChanged(level, pos, neighborBlock, orientation, movedByPiston);
             // Spigot start
         } catch (StackOverflowError ex) {
+            // Leaves start - update suppression crash fix
+            if (io.canvasmc.canvas.GlobalConfiguration.getInstance().vanillaFixes.updateSuppressionCrashFix) {
+                throw new io.canvasmc.canvas.util.UpdateSuppressionException(pos, level, neighborBlock, null, ex);
+            }
+            // Leaves end - update suppression crash fix
             level.lastPhysicsProblem = pos.immutable();
             // Spigot end
         } catch (Throwable var9) {
+            // Leaves start - update suppression crash fix
+            if (io.canvasmc.canvas.GlobalConfiguration.getInstance().vanillaFixes.updateSuppressionCrashFix) {
+                if (var9 instanceof io.canvasmc.canvas.util.UpdateSuppressionException ue) {
+                    ue.provideBlock(level, pos, neighborBlock);
+                    throw ue;
+                } else {
+                    throw new io.canvasmc.canvas.util.UpdateSuppressionException(pos, level, neighborBlock, null, var9);
+                }
+            }
+            // Leaves end - update suppression crash fix
             CrashReport crashReport = CrashReport.forThrowable(var9, "Exception while updating neighbours");
             CrashReportCategory crashReportCategory = crashReport.addCategory("Block being updated");
             crashReportCategory.setDetail(
