--- a/net/minecraft/resources/Identifier.java
+++ b/net/minecraft/resources/Identifier.java
@@ -23,6 +_,10 @@
     public static final String PAPER_NAMESPACE = "paper"; // Paper
     private final String namespace;
     private final String path;
+    // Canvas start - Cache identifier toString and hash
+    private String identifierStr;
+    private int hash;
+    // Canvas end - Cache identifier toString and hash
 
     private Identifier(String namespace, String path) {
         assert isValidNamespace(namespace);
@@ -130,7 +_,14 @@
 
     @Override
     public String toString() {
-        return this.namespace + ":" + this.path;
+        // Canvas start - Cache identifier toString and hash
+        String identifierStr = this.identifierStr;
+        if (identifierStr == null) {
+            identifierStr = this.namespace + ":" + this.path;
+            this.identifierStr = identifierStr;
+        }
+        return identifierStr;
+        // Canvas end - Cache identifier toString and hash
     }
 
     @Override
@@ -140,7 +_,14 @@
 
     @Override
     public int hashCode() {
-        return 31 * this.namespace.hashCode() + this.path.hashCode();
+        // Canvas start - Cache identifier toString and hash
+        int hash = this.hash;
+        if (hash == 0) {
+            hash = 31 * this.namespace.hashCode() + this.path.hashCode();
+            this.hash = hash;
+        }
+        return hash;
+        // Canvas end - Cache identifier toString and hash
     }
 
     @Override
