--- a/net/minecraft/advancements/criterion/BredAnimalsTrigger.java
+++ b/net/minecraft/advancements/criterion/BredAnimalsTrigger.java
@@ -18,7 +_,7 @@
     }
 
     public void trigger(ServerPlayer player, Animal parent, Animal partner, @Nullable AgeableMob child) {
-        if (io.canvasmc.canvas.Config.INSTANCE.disableCriterionTrigger) return; // Canvas - disable criterion trigger config
+        if (io.canvasmc.canvas.WorldConfig.getDefaults().disableCriterionTrigger) return; // Canvas - disable criterion trigger config
         LootContext lootContext = EntityPredicate.createContext(player, parent);
         LootContext lootContext1 = EntityPredicate.createContext(player, partner);
         LootContext lootContext2 = child != null ? EntityPredicate.createContext(player, child) : null;
