--- a/net/minecraft/advancements/criterion/CuredZombieVillagerTrigger.java
+++ b/net/minecraft/advancements/criterion/CuredZombieVillagerTrigger.java
@@ -17,7 +_,7 @@
     }
 
     public void trigger(ServerPlayer player, Zombie zombie, Villager villager) {
-        if (io.canvasmc.canvas.Config.INSTANCE.disableCriterionTrigger) return; // Canvas - disable criterion trigger config
+        if (io.canvasmc.canvas.WorldConfig.getDefaults().disableCriterionTrigger) return; // Canvas - disable criterion trigger config
         LootContext lootContext = EntityPredicate.createContext(player, zombie);
         LootContext lootContext1 = EntityPredicate.createContext(player, villager);
         this.trigger(player, instance -> instance.matches(lootContext, lootContext1));
