--- a/net/minecraft/advancements/criterion/ItemUsedOnLocationTrigger.java
+++ b/net/minecraft/advancements/criterion/ItemUsedOnLocationTrigger.java
@@ -30,7 +_,7 @@
     }
 
     public void trigger(ServerPlayer player, BlockPos pos, ItemStack stack) {
-        if (io.canvasmc.canvas.Config.INSTANCE.disableCriterionTrigger) return; // Canvas - disable criterion trigger config
+        if (io.canvasmc.canvas.WorldConfig.getDefaults().disableCriterionTrigger) return; // Canvas - disable criterion trigger config
         ServerLevel serverLevel = player.level();
         BlockState blockState = serverLevel.getBlockState(pos);
         LootParams lootParams = new LootParams.Builder(serverLevel)
