--- a/net/minecraft/advancements/criterion/KilledTrigger.java
+++ b/net/minecraft/advancements/criterion/KilledTrigger.java
@@ -17,7 +_,7 @@
     }
 
     public void trigger(ServerPlayer player, Entity entity, DamageSource damageSource) {
-        if (io.canvasmc.canvas.Config.INSTANCE.disableCriterionTrigger) return; // Canvas - disable criterion trigger config
+        if (io.canvasmc.canvas.WorldConfig.getDefaults().disableCriterionTrigger) return; // Canvas - disable criterion trigger config
         LootContext lootContext = EntityPredicate.createContext(player, entity);
         this.trigger(player, instance -> instance.matches(player, lootContext, damageSource));
     }
