--- a/net/minecraft/advancements/criterion/LightningStrikeTrigger.java
+++ b/net/minecraft/advancements/criterion/LightningStrikeTrigger.java
@@ -19,7 +_,7 @@
     }
 
     public void trigger(ServerPlayer player, LightningBolt lightning, List<Entity> nearbyEntities) {
-        if (io.canvasmc.canvas.Config.INSTANCE.disableCriterionTrigger) return; // Canvas - disable criterion trigger config
+        if (io.canvasmc.canvas.WorldConfig.getDefaults().disableCriterionTrigger) return; // Canvas - disable criterion trigger config
         List<LootContext> list = nearbyEntities.stream().map(entity -> EntityPredicate.createContext(player, entity)).collect(Collectors.toList());
         LootContext lootContext = EntityPredicate.createContext(player, lightning);
         this.trigger(player, instance -> instance.matches(lootContext, list));
