--- a/net/minecraft/advancements/criterion/TargetBlockTrigger.java
+++ b/net/minecraft/advancements/criterion/TargetBlockTrigger.java
@@ -17,7 +_,7 @@
     }
 
     public void trigger(ServerPlayer player, Entity projectile, Vec3 vector, int signalStrength) {
-        if (io.canvasmc.canvas.Config.INSTANCE.disableCriterionTrigger) return; // Canvas - disable criterion trigger config
+        if (io.canvasmc.canvas.WorldConfig.getDefaults().disableCriterionTrigger) return; // Canvas - disable criterion trigger config
         LootContext lootContext = EntityPredicate.createContext(player, projectile);
         this.trigger(player, instance -> instance.matches(lootContext, vector, signalStrength));
     }
