--- a/net/minecraft/advancements/criterion/TradeTrigger.java
+++ b/net/minecraft/advancements/criterion/TradeTrigger.java
@@ -17,7 +_,7 @@
     }
 
     public void trigger(ServerPlayer player, AbstractVillager villager, ItemStack stack) {
-        if (io.canvasmc.canvas.Config.INSTANCE.disableCriterionTrigger) return; // Canvas - disable criterion trigger config
+        if (io.canvasmc.canvas.WorldConfig.getDefaults().disableCriterionTrigger) return; // Canvas - disable criterion trigger config
         LootContext lootContext = EntityPredicate.createContext(player, villager);
         this.trigger(player, instance -> instance.matches(lootContext, stack));
     }
