--- a/net/minecraft/world/level/chunk/PalettedContainer.java
+++ b/net/minecraft/world/level/chunk/PalettedContainer.java
@@ -23,6 +_,22 @@
 import org.jspecify.annotations.Nullable;
 
 public class PalettedContainer<T> implements PaletteResize<T>, PalettedContainerRO<T> {
+
+    // Canvas start - Lithium - faster chunk serialization
+    private static final ThreadLocal<short[]> CACHED_ARRAY_4096 = ThreadLocal.withInitial(() -> new short[4096]);
+    private static final ThreadLocal<short[]> CACHED_ARRAY_64 = ThreadLocal.withInitial(() -> new short[64]);
+    private Optional<LongStream> asOptional(long[] data) {
+        return Optional.of(Arrays.stream(data));
+    }
+    private short[] getOrCreate(int size) {
+        return switch (size) {
+            case 64 -> CACHED_ARRAY_64.get();
+            case 4096 -> CACHED_ARRAY_4096.get();
+            default -> new short[size];
+        };
+    }
+    // Canvas end - Lithium - faster chunk serialization
+
     private static final int MIN_PALETTE_BITS = 0;
     public volatile PalettedContainer.Data<T> data; // Paper - optimise collisions - public
     public Strategy<T> strategy;
@@ -338,6 +_,56 @@
     public synchronized PalettedContainerRO.PackedData<T> pack(Strategy<T> strategy) { // Paper - synchronize
         this.acquire();
 
+        // Canvas start - Lithium - faster chunk serialization
+        if (io.canvasmc.canvas.GlobalConfiguration.getInstance().performance.fasterChunkSerialization) {
+            Optional<LongStream> data = Optional.empty();
+            List<T> elements = null;
+            try {
+                // The palette that will be serialized
+                io.canvasmc.canvas.lithium.common.world.chunk.LithiumHashPalette<T> hashPalette = null;
+
+                final Palette<T> palette = this.data.palette();
+                final BitStorage storage = this.data.storage();
+                if (storage instanceof ZeroBitStorage || palette.getSize() == 1) {
+                    // If the palette only contains one entry, don't attempt to repack it.
+                    elements = List.of(palette.valueFor(0));
+                } else if (palette instanceof io.canvasmc.canvas.lithium.common.world.chunk.LithiumHashPalette<T> lithiumHashPalette) {
+                    hashPalette = lithiumHashPalette;
+                }
+
+                if (elements == null) {
+                    io.canvasmc.canvas.lithium.common.world.chunk.LithiumHashPalette<T> compactedPalette = new io.canvasmc.canvas.lithium.common.world.chunk.LithiumHashPalette<>(storage.getBits());
+                    short[] array = this.getOrCreate(strategy.entryCount());
+
+                    storage.compact(this.data.palette(), compactedPalette, array, this);
+
+                    // If the palette didn't change during compaction, do a simple copy of the data array
+                    if (hashPalette != null && hashPalette.getSize() == compactedPalette.getSize() && storage.getBits() == strategy.getConfigurationForPaletteSize(hashPalette.getSize()).bitsInStorage()) { // paletteSize can de-sync from palette - see https://github.com/CaffeineMC/lithium-fabric/issues/279
+                        data = this.asOptional(storage.getRaw().clone());
+                        elements = hashPalette.getElements();
+                    } else {
+                        int bits = strategy.getConfigurationForPaletteSize(compactedPalette.getSize()).bitsInStorage();
+                        if (bits != 0) {
+                            // Re-pack the integer array as the palette has changed size
+                            SimpleBitStorage copy = new SimpleBitStorage(bits, array.length);
+                            for (int i = 0; i < array.length; ++i) {
+                                copy.set(i, array[i]);
+                            }
+
+                            // We don't need to clone the data array as we are the sole owner of it
+                            data = this.asOptional(copy.getRaw());
+                        }
+
+                        elements = compactedPalette.getElements();
+                    }
+                }
+            } finally {
+                this.release();
+            }
+
+            return new PalettedContainerRO.PackedData<>(elements, data);
+        }
+        // Canvas end - Lithium - faster chunk serialization
         PalettedContainerRO.PackedData var14;
         try {
             BitStorage bitStorage = this.data.storage;
@@ -410,6 +_,37 @@
 
     @Override
     public void count(PalettedContainer.CountConsumer<T> countConsumer) {
+        // Canvas start - Lithium - faster chunk serialization
+        if (io.canvasmc.canvas.GlobalConfiguration.getInstance().performance.fasterChunkSerialization) {
+            int len = this.data.palette().getSize();
+
+            // Do not allocate huge arrays if we're using a large palette
+            if (len > 4096) {
+                // VanillaCopy
+                if (this.data.palette.getSize() == 1) {
+                    countConsumer.accept(this.data.palette.valueFor(0), this.data.storage.getSize());
+                } else {
+                    Int2IntOpenHashMap map = new Int2IntOpenHashMap();
+                    this.data.storage.getAll(id -> map.addTo(id, 1));
+                    map.int2IntEntrySet().forEach(idEntry -> countConsumer.accept(this.data.palette.valueFor(idEntry.getIntKey()), idEntry.getIntValue()));
+                }
+                return;
+            }
+
+            short[] counts = new short[len];
+
+            this.data.storage().getAll(i -> counts[i]++);
+
+            for (int i = 0; i < counts.length; i++) {
+                T obj = this.data.palette().valueFor(i);
+
+                if (obj != null) {
+                    countConsumer.accept(obj, counts[i]);
+                }
+            }
+            return;
+        }
+        // Canvas end - Lithium - faster chunk serialization
         if (this.data.palette.getSize() == 1) {
             countConsumer.accept(this.data.palette.valueFor(0), this.data.storage.getSize());
         } else {
