--- a/net/minecraft/world/entity/ai/behavior/VillagerGoalPackages.java
+++ b/net/minecraft/world/entity/ai/behavior/VillagerGoalPackages.java
@@ -54,7 +_,7 @@
                     MemoryModuleType.POTENTIAL_JOB_SITE,
                     true,
                     Optional.empty(),
-                    (level, pos) -> true
+                    (level, pos) -> true, (body) -> body.level().canvasConfig().entities.villagers.reduceJobSitePoiSearchRange ? 16 : 48, (body) -> body.level().canvasConfig().entities.villagers.villagerAcquirePoiTasksLoadChunks // Canvas - configurable scan range and if poi should load
                 )
             ),
             Pair.of(7, new GoToPotentialJobSite(speedModifier)),
@@ -62,10 +_,10 @@
             Pair.of(
                 10,
                 AcquirePoi.create(
-                    poiType -> poiType.is(PoiTypes.HOME), MemoryModuleType.HOME, false, Optional.of((byte)14), VillagerGoalPackages::validateBedPoi
+                    poiType -> poiType.is(PoiTypes.HOME), MemoryModuleType.HOME, MemoryModuleType.HOME, false, Optional.of((byte)14), VillagerGoalPackages::validateBedPoi, (body) -> body.level().canvasConfig().entities.villagers.reduceHomePoiSearchRange ? 16 : 48, (body) -> body.level().canvasConfig().entities.villagers.villagerAcquirePoiTasksLoadChunks // Canvas - configurable scan range and if poi should load
                 )
             ),
-            Pair.of(10, AcquirePoi.create(poiType -> poiType.is(PoiTypes.MEETING), MemoryModuleType.MEETING_POINT, true, Optional.of((byte)14))),
+            Pair.of(10, AcquirePoi.create(poiType -> poiType.is(PoiTypes.MEETING), MemoryModuleType.MEETING_POINT, MemoryModuleType.MEETING_POINT, true, Optional.of((byte)14), (level, pos) -> true, (body) -> body.level().canvasConfig().entities.villagers.reduceMeetingPointPoiSearchRange ? 16 : 48, (body) -> body.level().canvasConfig().entities.villagers.villagerAcquirePoiTasksLoadChunks)), // Canvas - configurable scan range and if poi should load
             Pair.of(10, AssignProfessionFromJobSite.create()),
             Pair.of(10, ResetProfession.create())
         );
