--- a/io/papermc/paper/threadedregions/RegionShutdownThread.java
+++ b/io/papermc/paper/threadedregions/RegionShutdownThread.java
@@ -25,6 +_,7 @@
         this.setUncaughtExceptionHandler((thread, thr) -> {
             LOGGER.error("Error shutting down server", thr);
         });
+        this.setPriority(10); // Canvas - bump priority
     }
 
     static ThreadedRegionizer.ThreadedRegion<TickRegions.TickRegionData, TickRegions.TickRegionSectionData> getRegion() {
@@ -138,6 +_,7 @@
                         player.containerMenu.setCarried(ItemStack.EMPTY);
                         player.drop(carried, false);
                     }
+                    if (player.canvas$isTemporarilyBlocking) player.releaseUsingItem(); // Canvas - implement sword blocking
                 } catch (final Throwable thr) {
                     LOGGER.error("Failed to close player inventory for player: " + player, thr);
                 }
@@ -186,7 +_,7 @@
                     final RegionizedWorldData worldData = region.regioniser.world.worldRegionData.get();
                     for (final ServerPlayer localPlayer : worldData.getLocalPlayers()) {
                         // if pearl duplication fix is enabled, remove the pearl
-                        localPlayer.savePearls(io.canvasmc.canvas.Config.INSTANCE.fixes.pearlDuplication);
+                        localPlayer.savePearls(io.canvasmc.canvas.GlobalConfiguration.getInstance().vanillaFixes.pearlDuplication);
                     }
                 } catch (Throwable thrown) {
                     LOGGER.error("Failed to save pearls for region around chunk " + center + " in world '" + region.regioniser.world.getWorld().getName() + "'", thrown);
