--- a/io/papermc/paper/threadedregions/RegionizedTaskQueue.java
+++ b/io/papermc/paper/threadedregions/RegionizedTaskQueue.java
@@ -517,7 +_,7 @@
             }
 
             private ReferenceCountData getReferenceCounterVolatile() {
-                return (ReferenceCountData)REFERENCE_COUNTER_HANDLE.get(this);
+                return (ReferenceCountData)REFERENCE_COUNTER_HANDLE.getVolatile(this); // Canvas - Correct volatile reference get in RegionizedTaskQueue
             }
 
             private ReferenceCountData compareAndExchangeReferenceCounter(final ReferenceCountData expect, final ReferenceCountData update) {
