--- a/io/papermc/paper/entity/activation/ActivationRange.java
+++ b/io/papermc/paper/entity/activation/ActivationRange.java
@@ -231,7 +_,7 @@
         }
         // special cases.
         if (entity instanceof final LivingEntity living) {
-            if (living.onClimbable() || living.jumping || living.hurtTime > 0 || !living.activeEffects.isEmpty() || living.isFreezing()) {
+            if (living.blockPosition().equals(living.getLastClimbablePos().orElse(null)) || living.jumping || living.hurtTime > 0 || !living.activeEffects.isEmpty() || living.isFreezing()) { // Canvas - backport Paper#12764 from 26.1
                 return 1;
             }
             if (entity instanceof final Mob mob && mob.getTarget() != null) {
@@ -299,7 +_,7 @@
      */
     public static boolean checkIfActive(final Entity entity) {
         // Never safe to skip fireworks or item gravity
-        if (entity instanceof FireworkRocketEntity || (entity instanceof ItemEntity && (entity.tickCount + entity.getId()) % 4 == 0)) { // Needed for item gravity, see ItemEntity tick
+        if (entity instanceof FireworkRocketEntity || ((entity instanceof ItemEntity && !entity.isInWater() && (entity.tickCount + entity.getId()) % 4 == 0))) { // Needed for item gravity, see ItemEntity tick // Canvas - fix ItemEntity abnormal movement in water
             return true;
         }
         // special case always immunities
