--- a/net/minecraft/world/entity/vehicle/boat/AbstractBoat.java
+++ b/net/minecraft/world/entity/vehicle/boat/AbstractBoat.java
@@ -188,25 +_,25 @@
         if (entity instanceof AbstractBoat) {
             if (entity.getBoundingBox().minY < this.getBoundingBox().maxY) {
                 // CraftBukkit start
-                if (!this.isPassengerOfSameVehicle(entity)) {
+                if (!this.isPassengerOfSameVehicle(entity) && org.bukkit.event.vehicle.VehicleEntityCollisionEvent.getHandlerList().getRegisteredListeners().length > 0) { // Canvas - Skip VehicleEntityCollisionEvent if no listeners
                     org.bukkit.event.vehicle.VehicleEntityCollisionEvent event = new org.bukkit.event.vehicle.VehicleEntityCollisionEvent(
                         (org.bukkit.entity.Vehicle) this.getBukkitEntity(),
                         entity.getBukkitEntity()
                     );
                     if (!event.callEvent()) return;
-                }
+                } // Canvas - Skip VehicleEntityCollisionEvent if no listeners
                 // CraftBukkit end
                 super.push(entity);
             }
         } else if (entity.getBoundingBox().minY <= this.getBoundingBox().minY) {
             // CraftBukkit start
-            if (!this.isPassengerOfSameVehicle(entity)) {
+            if (!this.isPassengerOfSameVehicle(entity) && org.bukkit.event.vehicle.VehicleEntityCollisionEvent.getHandlerList().getRegisteredListeners().length > 0) { // Canvas - Skip VehicleEntityCollisionEvent if no listeners
                 org.bukkit.event.vehicle.VehicleEntityCollisionEvent event = new org.bukkit.event.vehicle.VehicleEntityCollisionEvent(
                     (org.bukkit.entity.Vehicle) this.getBukkitEntity(),
                     entity.getBukkitEntity()
                 );
                 if (!event.callEvent()) return;
-            }
+            } // Canvas - Skip VehicleEntityCollisionEvent if no listeners
             // CraftBukkit end
             super.push(entity);
         }
@@ -282,7 +_,9 @@
 
         if (this.lastLocation != null && !this.lastLocation.equals(to)) {
             org.bukkit.event.vehicle.VehicleMoveEvent event = new org.bukkit.event.vehicle.VehicleMoveEvent(vehicle, this.lastLocation, to);
+            this.blockTeleportAsync = true; // Canvas - Prevent teleportAsync calls during move events
             event.callEvent();
+            this.blockTeleportAsync = false; // Canvas - Prevent teleportAsync calls during move events
         }
         this.lastLocation = vehicle.getLocation();
         // CraftBukkit end
@@ -786,7 +_,7 @@
 
     @Override
     protected boolean canAddPassenger(Entity passenger) {
-        return this.getPassengers().size() < this.getMaxPassengers() && !this.isEyeInFluid(FluidTags.WATER);
+        return this.getPassengers().size() < this.getMaxPassengers() && !this.canvas$isEyeInWater(); // Canvas - optimize fluidOnEyes
     }
 
     protected int getMaxPassengers() {
